### PR TITLE
ESLint Reconfiguration and Related Fixes

### PR DESCRIPTION
**Pull request by jperson1**
# ESLint Reconfiguration and Related Fixes

Issue: https://github.com/GSA-TTS/FAC/issues/3933

## Changes:
1. Have a standard `npx eslint` cover all the files we need covered
    * Update `eslint.config.mjs` to handle standard options, such as ignored directories
    * Update the package scripts and Makefile to use the standard execution
    * Now covers Cypress, it's own config file, etc.
2. Migrate a number of files to ES modules format
    * ES modules have been supported by node since 2017, so we're good there
    * Cypress config and plugin index files are updates, as well as the ESBuild specific `build-assets.mjs`
    * _The future is now_
3. Fix all the ESLint errors. 
    * Most are a Cypress error akin to `It is unsafe to chain further commands that rely on the subject after this command.` This means we shouldn't clear a text box, type into it, and blur it in rapid succession. These just get broken into several statements where re recapture the text box each time to verify that it's okay. 
    * A bunch of ES module changes
    * That's pretty much it? We already used prettier for just about everything, so there's more actual errors/warnings than formatting errors. 

Also, I included the ESLint Babel parser (@babel/eslint-parser). ESLint only officially supports the latest final ECMAScript standard, so Babel helps keep everything non-ECMA in check.

## How to test:
Locally:
1. Switch to this branch and run normally.
2. Bring up a fresh stack, including refreshing the `node_modules` directory and running a `collectstatic`
3. Run `npx eslint` to verify the new config works.
5. Run the Cypress tests, to verify that the not-insignificant changes didn't change the behavior.

Other things:
1. The GitHub Actions PR tests should still pass.
2. Cypress through GitHub Actions should also still pass.

## PR checklist: submitters

- [x]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [x]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [x]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [x]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [x]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [x]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [x]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [x]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

**Comment by jperson1** [comment]
This Cypress run shows that these changes didn't break the Cypress config: https://github.com/GSA-TTS/FAC/actions/runs/9455963528

They're failing, but it's likely because they were run against dev, which is likely not equipped to handle the authorization pieces. The important part is that they load up and start the tests with no issue.

File: backend/cypress.config.mjs [hunk 1]
```diff
@@ -1,13 +1,14 @@
-const { defineConfig } = require('cypress');
-const { downloadFile } = require('cypress-downloadfile/lib/addPlugin');
+import { defineConfig } from 'cypress';
+import { downloadFile } from 'cypress-downloadfile/lib/addPlugin.js';
+import indexTasks from './cypress/plugins/index.mjs';
 
-module.exports = defineConfig({
+export default defineConfig({
```
Comment (danswick) [inline backend/cypress.config.mjs:5]: 🙏

File: backend/cypress/e2e/check-eligibility.cy.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ describe('Create New Audit', () => {
     });
 
     it('should navigate to next step after successful submission', () => {
-      testValidEligbility();
+      testValidEligibility();
```
Comment (danswick) [inline backend/cypress/e2e/check-eligibility.cy.js:81]: Nice catch. This one was hard to spot, even in the diff.

File: backend/cypress/e2e/check-ueid.cy.js [hunk 3]
```diff
@@ -39,22 +40,28 @@ describe('Create New Audit', () => {
...
-        cy.get('#auditee_uei').type('ASDF').blur();
+        cy.get('#auditee_uei').type('ASDF');
+        cy.get('#auditee_uei').blur();
         cy.get('#auditee_uei-length').should('be.visible');
       });
 
       it('should indicate when the supplied input is too long', () => {
-        cy.get('#auditee_uei').clear().type('ASDFASDFASDFA').blur();
+        cy.get('#auditee_uei').clear();
+        cy.get('#auditee_uei').type('THIRTEENLONG0');
```
Comment (danswick) [inline backend/cypress/e2e/check-ueid.cy.js:56]: :heart:

File: backend/package.json [hunk 4]
```diff
@@ -9,10 +9,10 @@
...
   "scripts": {
     "build": "run-p build:*",
-    "build:esbuild": "node build-assets.js",
-    "dev": "node build-assets.js --watch",
-    "check:eslint": "npx eslint --ignore-pattern ../.gitignore static/js",
-    "fix:eslint": "npx eslint --fix --ignore-pattern ../.gitignore static/js",
+    "build:esbuild": "node build-assets.mjs",
+    "dev": "node build-assets.mjs --watch",
+    "check:eslint": "npx eslint",
+    "fix:eslint": "npx eslint --fix",
```
Comment (danswick) [inline backend/package.json:15]: I'm not sure why they were like this, but we don't need `npx` to run these commands as npm scripts. 

```suggestion
    "check:eslint": "eslint",
    "fix:eslint": "eslint --fix",
```

**Review by danswick — APPROVED** [review]
This looks great. Thanks, @jperson1! I left one small suggestion inline. Feel free to incorporate if you agree or merge if you don't.

File: backend/package.json (hunk 4 above)
Comment (jperson1) [reply backend/package.json]: Oh, fantastic! Went ahead and axed the `npx`'s for the other scripts as well.

**Review by danswick — APPROVED** [review]
Looks good to go!

**Pull request merged**